modelkit/assets/drivers/azure.py

```diff
@@ -23,18 +23,28 @@ def __init__(
         connection_string: Optional[str] = None,
         client: Optional[BlobServiceClient] = None,
     ):
-        self.bucket = bucket or os.environ.get("MODELKIT_STORAGE_BUCKET") or ""
-        if not self.bucket:
-            raise ValueError("Bucket needs to be set for Azure storage driver")
+        client_configuration = {
+            "connection_string": connection_string
+            or os.environ.get("AZURE_STORAGE_CONNECTION_STRING")
+        }
+        if not (client or client_configuration["connection_string"]):
+            raise ValueError(
+                "Connection string needs to be set for Azure storage driver"
+            )
+        super().__init__(
+            bucket=bucket, client=client, client_configuration=client_configuration
+        )
 
-        if client:
-            self.client = client
-        elif connection_string:  # pragma: no cover
-            self.client = BlobServiceClient.from_connection_string(connection_string)
-        else:
-            self.client = BlobServiceClient.from_connection_string(
-                os.environ["AZURE_STORAGE_CONNECTION_STRING"]
+    @staticmethod
+    def build_client(
+        client_configuration: Union[Dict[str, Any], Optional[Dict[str, Any]]]
```

Comment:
NB: just `Optional[Dict[str, Any]]` does not fit ?

Reply:
I have narrowed it down to a simple `Dict[str, Any]`, thanks for the heads up @tgenin